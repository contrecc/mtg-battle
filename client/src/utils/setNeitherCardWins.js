export default function setNeitherCardWins() {
  return function(state, props) {
    return {
      bothWeak: true,
      winStreak: state.winStreak + 1,
      winningCard: '',
      winner: false,
      roundCompleted: true
    };
  };
}
