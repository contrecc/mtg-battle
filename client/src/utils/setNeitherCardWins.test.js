import setNeitherCardWins from './setNeitherCardWins';

describe('setNeitherCardWins', () => {
  it('should return a function', () => {
    expect(setNeitherCardWins()).toMatchSnapshot();
  });

  it('should return a function that then updates the state', () => {
    const returnedFunction = setNeitherCardWins();
    expect(returnedFunction({ winStreak: 1 }, {})).toMatchSnapshot();
  });
});
