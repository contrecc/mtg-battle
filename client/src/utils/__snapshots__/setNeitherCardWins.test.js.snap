// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setNeitherCardWins should return a function 1`] = `[Function]`;

exports[`setNeitherCardWins should return a function that then updates the state 1`] = `
Object {
  "bothWeak": true,
  "roundCompleted": true,
  "winStreak": 2,
  "winner": false,
  "winningCard": "",
}
`;
